src/OpenTracing/Basic/Propagator.php

```diff
@@ -0,0 +1,27 @@
+<?php
+
+namespace OpenTracing\Basic;
+
+use OpenTracing;
+use OpenTracing\Extractor;
+use OpenTracing\Injector;
+
+abstract class Propagator implements Injector, Extractor {
+
+	const FIELD_STATE = 'state';
+	const FIELD_ATTRIBUTES = 'attributes';
+	const FIELD_TRACE_ID = 'traceid';
+	const FIELD_SPAN_ID = 'spanid';
+
+	protected $tracer = null;
+
+	public function __construct( Tracer $tracer ) {
+		$this->tracer = $tracer;
+	}
+
+	protected function validateSpan( OpenTracing\Span $span ) {
+		if ( !$span instanceof Span ) {
```

Comment:
Unfortunately it would break the OO compatibility between `OpenTracing\Span` interface and the particular implementation eg. this one `OpenTracing\Basic\Span`. Thus we need to do this check at runtime. 
However let me know if you have other idea for solving this compatibility issue.